pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableUtils.java

```diff
@@ -107,6 +108,34 @@ public static DataTable buildEmptyDataTable(QueryContext queryContext)
     }
   }
 
+  public static DataTable buildMetadataOnlyDataTable(DataTable origDataTable) {
```

Comment:
I feel we might better handle this in the table table, instead of replicating all the metadata keys. Handling it this way can add maintenance overhead

Reply:
you meant in the serialization?

Reply:
As `DataTable.toMetadataOnlyDataTable()`

Reply:
doing this without replicating is dangerous unless we make the resulting `MetadataOnlyDataTable` and `DataOnlyDataTable` somehow immutable 

e.g. you have additional object reference outside of the original DataTable that can mutate the content of itself. This can lead to unexpected behaviors

Reply:
i can move this into DataTable as `dataTable.toMetadataOnlyDataTable()` but I will still replicate and produce a copy. is this what you had in mind?

Reply:
The data in `DataTable` is actually immutable, only the metadata is mutable. For `toMetadataOnlyDataTable()`, we create a new data table and only keep the metadata; for `toDataOnlyDataTable()`, we create a new data table with all the data fields but replace the metadata map with an empty map. wdyt?

Reply:
oh! sounds good. if we guarantee that the data inside `DataTable` is immutable. I can definitely go with this path!